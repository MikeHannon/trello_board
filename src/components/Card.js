import React, { PropTypes } from 'react';
import './card.css';

const Card = ({status, title, description}) => {
  return (
    <div className={`card ${status}`}>
      <header>
        <h1>{title}</h1>
        <p>{description}</p>
      </header> 
    </div>
  )
}

Card.propTypes = {
  status: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
}

export default Card;