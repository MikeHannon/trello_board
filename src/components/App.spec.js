import React from 'react'
import Card from './Card'
import renderer from 'react-test-renderer'
import { DragDropContext } from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import data from '../../card_data.json'
const WrappedCard = DragDropContext(HTML5Backend)(Card)

test('name of test', () => {
    const component = renderer.create(<WrappedCard {...data.cards[0]}></WrappedCard>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
