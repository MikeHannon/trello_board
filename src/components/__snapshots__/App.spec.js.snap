exports[`test name of test 1`] = `
<div
  className="card complete">
  <header>
    <h1>
      Build this app
    </h1>
    <p>
      Use drag n\' drop behavior and CSS modules
    </p>
  </header>
</div>
`;
